src/components/Form.js

```diff
@@ -1,20 +1,53 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { useDispatch } from 'react-redux';
+import { bookAdded } from '../Redux/books/bookSlice';
 
-const Form = () => (
-  <>
-    <form className="form">
-      <input type="text" name="title" placeholder="Book title" />
-      <select className="form-select">
-        <option value="author">Select Author</option>
-        <option value="option-1">Select option 1</option>
-        <option value="option-2">Select option 2</option>
-        <option value="option-3">Select option 3</option>
-      </select>
-      <button className="form-btn" type="submit">
-        Add Book
-      </button>
-    </form>
-  </>
-);
+const Form = () => {
+  const dispatch = useDispatch();
+  const [state, setState] = useState({ id: '', title: '', author: '' });
+
+  const addBookToStore = (e) => {
+    e.preventDefault();
+    if (state.title === '') {
+      return;
+    }
+    dispatch(bookAdded(state));
+    setState({ id: '', title: '', author: '' });
+  };
+
+  const handleTitleChange = (e) => {
+    setState({
+      ...state,
+      id: uuidv4(),
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  return (
+    <>
+      <form className="form" onSubmit={addBookToStore}>
+        <input
+          type="text"
+          name="title"
+          placeholder="Book title"
+          value={state.title}
+          onChange={handleTitleChange}
+        />
```

Comment:
- [ ] When the book data is submitted, the books don't get rendered on the screen because on the page you're rendering the books, you're not rendering the books from the store. One can't tell if the data is stored in the redux store. Kindly work on that 👍🏽

Reply:
Hello @AmaduKamara Thank you for your help! but I think that I submitted the wrong branch or something because the correct changes doesn't appear here. going to submit again.